### PR TITLE
Yasara branch 2

### PR DESCRIPTION
**Pull request by YasaraPeiris**
Second pull request - add event, edit event, add stream, edit stream, delete stream

### CODE REVIEW

File: bootstrap-vue-web/src/assets/css/style.css [hunk 1]
```diff
@@ -1,17 +1,65 @@
...
+.btn-danger{
+  background-color: #bd1727;
+  border: 1px solid #9d0717;
+}
+.jsoneditor{
+  border: thin solid darkgray !important;
+}
+
+.jsoneditor-menu
+{
```
Comment (Zebedeusz) [inline bootstrap-vue-web/src/assets/css/style.css:58]: Format this file

File: bootstrap-vue-web/src/components/layout/SideBar.vue [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div style="max-width: 450px;min-width: 200px;">
+  <div class="max-width">
```
Comment (Zebedeusz) [inline bootstrap-vue-web/src/components/layout/SideBar.vue:2]: Try applying smth of this kind: http://getbem.com/naming/
E.g. in this kind, the class could be named simply `sidebar`

File: bootstrap-vue-web/src/components/login/ServiceInfo.vue [hunk 3]
```diff
@@ -4,34 +4,34 @@
...
+      </div>
     </div>
   </div>
 </template>
 
 <script>
-import PryvInput from "../../shared/PryvInput";
-import { constants } from "../../../utilities/constants";
+import PryvInput from "../shared/PryvInput";
+import { constants } from "../../utilities/constants";
```
Comment (Zebedeusz) [inline bootstrap-vue-web/src/components/login/ServiceInfo.vue:34]: You can use paths like: `@/utilities/...`

File: bootstrap-vue-web/src/components/modals/AddStreamModal.vue [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    @hidden="resetModal"
+    @ok="handleOk"
+    @show="resetModal"
+  >
+    <template v-slot:modal-header>
+      <h5>Add a new stream</h5>
+    </template>
+    <template v-slot:default>
+      <form ref="form" @submit.stop.prevent="handleSubmit">
+        <b-form-group label="ID" label-for="isnput-id">
```
Comment (Zebedeusz) [inline bootstrap-vue-web/src/components/modals/AddStreamModal.vue:14]: Typo

File: bootstrap-vue-web/src/components/modals/AddStreamModal.vue [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+      return this.streamsMap[filteredObj].map(obj => ({
+        value: `${obj.name} [${obj.id}]`,
+        text: filteredObj,
+      }));
+    },
+  },
+  methods: {
+    checkFormValidity() {
+      const valid = this.$refs.form.checkValidity();
+      if (this.selectedName === null || this.selectedName === "")
```
Comment (Zebedeusz) [inline bootstrap-vue-web/src/components/modals/AddStreamModal.vue:129]: try: `if(this.selectedName)`

File: bootstrap-vue-web/src/components/modals/AddStreamModal.vue [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+        } else {
+          await this.addStreamsToStore(result[0].stream);
+        }
+      } catch (e) {
+        console.log("Error occurred when creating modals" + e);
+
+      }
+    },
+    async addStreamsToStore(stream) {
+      let clonedStreams = JSON.parse(JSON.stringify(this.streamsMap));
```
Comment (Zebedeusz) [inline bootstrap-vue-web/src/components/modals/AddStreamModal.vue:174]: Why not? `Object.assign({}, obj);`
https://stackoverflow.com/questions/122102/what-is-the-most-efficient-way-to-deep-clone-an-object-in-javascript

File: bootstrap-vue-web/src/components/shared/Checkbox.vue [hunk 7]
```diff
@@ -2,11 +2,11 @@
   <span>
     <input
       :checked="checked"
-      name="checkbox-1"
       @change="$emit('change', $event.target.checked)"
+      name="checkbox-1"
       type="checkbox"
     />
-    <label>
+    <label class="label-class">
```
Comment (Zebedeusz) [inline bootstrap-vue-web/src/components/shared/Checkbox.vue:9]: You can name it just `label`

File: bootstrap-vue-web/src/components/shared/Checkbox.vue [hunk 8]
```diff
@@ -30,10 +30,10 @@ export default {
 };
 </script>
 
-<style scoped>
-label {
+<style>
```
Comment (Zebedeusz) [inline bootstrap-vue-web/src/components/shared/Checkbox.vue:33]: Be careful with unscoped styles

File: bootstrap-vue-web/src/components/shared/GroupCheckBox.vue [hunk 9]
```diff
@@ -61,6 +71,16 @@ export default {
           ] = this.computedAccessInfoObjectArray[e.endpointClicked].map(opt => {
             return opt.streamId;
           });
+          clonedSelectedStreamsObjectArray[e.endpointClicked].forEach(obj => {
```
Comment (Zebedeusz) [inline bootstrap-vue-web/src/components/shared/GroupCheckBox.vue:74]: ...ObjectArray - find a different name for this variable

File: bootstrap-vue-web/src/components/shared/RecursiveCheckbox.vue [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+        <b-icon-pencil-square
+          @click="viewStreamInfoRecursive(stream.streamId, endpoint, $event)"
+          class="pencil-btn"
+        >
+        </b-icon-pencil-square>
+      </b-col>
+      <b-col cols="12">
+        <b-collapse :id="`${stream.streamId}`" class="mt-2">
+          <b-card>
+            <RecursiveCheckbox
```
Comment (Zebedeusz) [inline bootstrap-vue-web/src/components/shared/RecursiveCheckbox.vue:49]: It contains itself?

File: bootstrap-vue-web/src/components/shared/StreamCheckBox.vue [hunk 11]
```diff
@@ -81,11 +131,26 @@ export default {
       };
       this.$emit("checkBoxClicked", payload);
     },
-    viewAccessInfo(val) {
-      if (this.currentRouteName != "Info") {
+    viewAccessInfoFunc(endpoint) {
```
Comment (Zebedeusz) [inline bootstrap-vue-web/src/components/shared/StreamCheckBox.vue:134]: Why `Func` at the end?

File: bootstrap-vue-web/src/components/shared/RecursiveCheckbox.vue (hunk 10 above)
Comment (YasaraPeiris) [reply bootstrap-vue-web/src/components/shared/RecursiveCheckbox.vue]: yes this is recursive

**Pull request merged**